fs/parseduration.go

```diff
@@ -231,10 +232,30 @@ func (d Duration) Type() string {
 
 // UnmarshalJSON makes sure the value can be parsed as a string or integer in JSON
 func (d *Duration) UnmarshalJSON(in []byte) error {
-	return UnmarshalJSONFlag(in, d, func(i int64) error {
-		*d = Duration(i)
+	// Check if the input is a string value.
+	if in[0] == '"' {
```

Comment:
You should check `len(in) > 0` before checking `in[0]`.  In fact for this if branch you want `len(in) >= 2` I think.

So I think this should say

```go
if len(in) >= 2 && in[0] == '"' && in[len(in)-1] == '"' {
```